core/types/transaction.go

```diff
@@ -415,6 +415,22 @@ func (tx *Transaction) Copy() *Transaction {
 	return &tx2
 }
 
+// SenderAddress returns the address of transaction sender
+// Note that mainnet has unprotected transactions prior to Epoch 28
+func (tx *Transaction) SenderAddress() (common.Address, error) {
+	var signer Signer
+	if !tx.Protected() {
+		signer = HomesteadSigner{}
+	} else {
+		signer = NewEIP155Signer(tx.ChainID())
+	}
+	addr, err := Sender(signer, tx)
+	if err != nil {
+		return common.Address{}, err
```

Comment:
please first one becomes pointer and nil. If giving back non-nil error, then proper value should be nil

Reply:
To be consistent with staking tx, I think it should be as-is: 

https://github.com/harmony-one/harmony/blob/master/staking/types/transaction.go#L314  

It would require a lot more change on things that consume staking tx sender addr if we decide to change it the other way IMO. Not sure if worth it.